src/engine/tests/engine/RelativeUpdateTest.cpp

```diff
@@ -0,0 +1,72 @@
+#include <gtest/gtest.h>
+
+#include <array>
+#include <chrono>
+
+#include "Entity.hpp"
+#include "Registry.hpp"
+#include "RelativeTimeUpdate.hpp"
+
+using namespace ES::Engine;
+using namespace std::chrono_literals;
+
+TEST(Registry, RelativeTimeUpdateBasic)
+{
+    Registry reg;
+
+    int update_count = 0;
+
+    float deltaTime1 = 0.0f;
+    float deltaTime2 = 0.0f;
+
+    // Relative time update uses the highest precision available given how often it is called
+    // it only uses multiple updates if the time difference is greater than the target tick rate
+    reg.RegisterSystem<Scheduler::RelativeTimeUpdate>([&](Registry &registry) {
+        if (deltaTime1 == 0.0f)
+        {
+            deltaTime1 = reg.GetScheduler<Scheduler::RelativeTimeUpdate>().GetCurrentDeltaTime();
```

Comment:
This test make me think about that we will not call other schedulers than current one inside systems (maybe I'm wrong) so it should be better to have a method to get current scheduler rather than specifiying the type of scheduler each time. This can also enhance the portability of systems: if we want to run a system inside a different scheduler, it will not create bugs or error due to the switch.

Reply:
Better namming and portability of systems but yeah, for the moment it's fine